Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,488 @@
+  let T := max B 1
+  have h_w_pos (z : ℂ) (hset : z.re ∈ Icc 0 (1 : ℝ)) (him : T ≤ |z.im|) : z ∈ w :=
+    h_w z hset (le_trans (le_max_left B 1) him)
+
+  by_cases (|z.im| ≤ T)
+  -- First case |z.im| ≤ T
+  · have bdd_strip_is_bounded : Bornology.IsBounded (bddStrip 0 1 (T)) :=
+      ((Metric.isBounded_Ioo _ _).reProdIm (Metric.isBounded_Ioo _ _))
+
+    --Function is DiffContOnCl on subset + multiplied by 'simple' function
+    have hd_subset : DiffContOnCl ℂ (F' n f) (bddStrip 0 1 (T)) := by
+      apply DiffContOnCl.mono _ _
+      · use strip 0 1
+      · exact F'_diff_cont _ _ hd
+      · exact inter_subset_left _ _
+
+    apply norm_le_of_forall_mem_frontier_norm_le bdd_strip_is_bounded hd_subset _ _
+
+    -- Frontier bounded by one
+    · intros z hfz
+      rw [bddStrip, ← Set.reProdIm, frontier_reProdIm] at hfz
+      cases' hfz with hfz1 hfz2
+      · cases' hfz1 with hfz_re hfz_im
+        rw [frontier_Ioo] at hfz_im
+        rw [closure_Ioo (zero_ne_one' ℝ)] at hfz_re
+        apply h_h; apply h_w_pos z hfz_re
+        -- |T| = |z.im|
+        apply le_of_eq
+        rwa [eq_comm, abs_eq (le_trans (zero_le_one' ℝ) (le_max_right B 1)),
+         ← mem_singleton_iff, or_comm, ← mem_insert_iff, ← mem_preimage]
+        -- -T < T
+        exact neg_lt_self_iff.mpr (lt_of_lt_of_le (zero_lt_one' ℝ) (le_max_right B 1))
+      · replace hfz_re := mem_of_mem_inter_left hfz2
+        rw [frontier_Ioo (zero_lt_one' ℝ)] at hfz_re
+        exact edges_le_one z f n hB hfz_re
+    -- Local goal: z ∈ closure (bddStrip 0 1 T) with h : |z.im| ≤ T and hz : z ∈ strip 0 1.
+    · rw [bddStrip, ← Set.reProdIm, closure_reProdIm, closure_Ioo (zero_ne_one' ℝ), closure_Ioo,
+      Set.reProdIm, mem_inter_iff]
+      refine ⟨hz, ?_⟩
+      rwa [mem_preimage, mem_Icc, ← abs_le]
+      apply ne_of_lt
+      exact neg_lt_self_iff.mpr (lt_of_lt_of_le (zero_lt_one' ℝ) (le_max_right B 1))
+
+  -- Now : T < |z.im|.
+  · simp only [not_le] at h; apply h_h; exact h_w_pos z hz (le_of_lt h)
+
+--Proof that F_seq Tendsto F
+lemma F_seq_to_F (f : ℂ → ℂ) (z : ℂ) : Tendsto (fun n : ℕ ↦ F' n f z ) atTop (nhds (F f z)) :=
+  have mul_const : Tendsto (fun n : ℕ ↦ (z^2-1) * (n : ℝ)⁻¹) atTop (nhds 0) := by
+    simpa only [mul_zero]
+      using tendsto_const_nhds.mul (tendsto_algebraMap_inverse_atTop_nhds_0_nat ℂ)
+
+  have comp_exp : Tendsto (fun n : ℕ ↦ exp ( (z^2-1) * (n : ℝ)⁻¹)) atTop (nhds 1) := by
+    simpa only [exp_zero]
+      using  (Continuous.tendsto continuous_exp 0).comp mul_const
+
+  by simpa only [mul_one]
+    using tendsto_const_nhds.mul comp_exp
+
+-- Proof that |F_seq| Tendsto |F|
+lemma F_seq_to_F_abs (f : ℂ → ℂ) (z : ℂ) :
+    Tendsto (fun n : ℕ ↦ (abs ∘ (F' n f)) z ) atTop (nhds ((abs ∘ (F f)) z)) :=
+  (Continuous.tendsto continuous_abs (F f z)).comp (F_seq_to_F f z)
+
+
+-- We are now ready to combine of Hadamard_sequence with F_seq_to_F_abs:
+
+/--
+Hadamard three-line theorem on `[0,1]`: If `f` is a bounded function, continuous on `[0,1]`
```

Comment:
```suggestion
**Hadamard three-line theorem** on `[0,1]`: If `f` is a bounded function, continuous on `[0,1]`
```
We usually do this for theorems that actually have a name!

Reply:
Done